scipy/cluster/hierarchy.py

```diff
@@ -2575,40 +2571,48 @@ def fcluster(Z, t, criterion='inconsistent', depth=2, R=None, monocrit=None):
     all data points to be merged together - so a single cluster is returned.
 
     """
-    Z = np.asarray(Z, order='c')
+    xp = array_namespace(Z)
+    Z = as_xparray(Z, order='c', xp=xp)
     is_valid_linkage(Z, throw=True, name='Z')
 
+    t = xp.asarray(t)
```

Comment:
This should not be needed, and the `t_ = xp.astype(...)` doesn't seem right. This is documented as a scalar, and that's what the Cython code expects. It's unclear to me why this code wouldn't work without these changes.